lib/postgres_ext/serializers/active_model/array_serializer.rb

```diff
@@ -145,8 +141,25 @@ def _process_has_many_relation(key, association_reflection, relation_query, ids_
       { table: cte_name, ids_column: id_column_name, foreign_key: association_reflection.foreign_key }
     end
 
-    def _visitor
-      @_visitior ||= object.klass.connection.visitor
+    def _to_sql(arel, bind_values = nil)
+      @_visitor ||= object.klass.connection.visitor
+      if @_visitor.method(:accept).arity == 2
+        collector = Arel::Collectors::SQLString.new
+        collector = @_visitor.accept arel, collector
+        res = collector.value
+        unless bind_values.nil?
+          bind_values.each_with_index do |bv, i|
+            if bv[1].is_a? String
+              res = res.gsub(/\$#{i+1}(\b|\Z)/, "'#{bv[1].gsub("'", "''")}'")
+            else
```

Comment:
The context of this commit is from #14 and https://github.com/crossroads/postgres_ext-serializers/commit/0121fad6c14cf80b6ecaa9fc64b3d8e136b08113. It was a quick fix for replacing parametrized values (i.e. bind values) in a sql string e.g. the {0} in `select * from table where column = {0}`, using string manipulation without spending time investigating how to do this properly using Arel or seeing how this normally works in ActiveModel. And yes here I'm merely testing if the value is a string and then wrapping it in quotes and escaping quotes within the value (which from my understanding stop sql injection attacks and malformed sql errors) before inserting, otherwise just inserting the value as is. But looking at this now I already see I've missed testing if the value is a symbol which would allow a sql injection attack. So I'd vote for looking into this properly before inclusion in postgres_ext-serializers, but unfortunately I'm quite busy at the moment maybe in a few weeks time I can take another look.